airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/avro/AvroSerializedBuffer.kt

```diff
@@ -102,7 +103,7 @@ class AvroSerializedBuffer(
                             .filter { s: ConfiguredAirbyteStream ->
                                 s.stream.name == stream.name &&
                                     StringUtils.equals(
-                                        s.stream.namespace,
+                                        s.stream.namespace ?: defaultNamespace,
```

Comment:
I really hate how we have to have those all over the place...

Reply:
Same. The issue is that the async framework mangles the namespace, so I needed to adapt everything downstream of that to recognize `""` as `null`. It's ugly and brittle, but the alternative is to try to undo the mangling in the async consumer, which is frightening.

Reply:
What's especially ugly about it is that it's threaded through as a parameter, but if it were set to anything other than `""` I think it might break.